src/routes/new-open/Open.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactElement } from 'react'
+import IconButton from '@material-ui/core/IconButton'
+import ChevronLeft from '@material-ui/icons/ChevronLeft'
+import { makeStyles } from '@material-ui/core/'
+
+import Page from 'src/components/layout/Page'
+import Block from 'src/components/layout/Block'
+import Row from 'src/components/layout/Row'
+import Heading from 'src/components/layout/Heading'
+import { history } from 'src/store'
+import { sm } from 'src/theme/variables'
+import StepperForm, { StepFormElement } from 'src/components/StepperForm/StepperForm'
+import SelectNetworkStep, { selectNetworkStepLabel } from 'src/components/SelectNetworkStep/SelectNetworkStep'
+import NameNewSafeStep, { nameNewSafeStepLabel } from './steps/NameNewSafeStep'
+import { APP_ENV } from 'src/utils/constants'
+import { FIELD_CREATE_SUGGESTED_SAFE_NAME } from './fields/createSafeFields'
+import { getRandomName } from 'src/logic/hooks/useMnemonicName'
+
+function Open(): ReactElement {
+  const classes = useStyles()
+
+  function onSubmitCreateNewSafe(values) {
+    // TODO: onSubmitCreateNewSafe
+    // TODO: Safe Creation Process Component
+    console.log('onSubmitCreateNewSafe', values)
+  }
+
+  const isProductionEnv = APP_ENV === 'production'
+
+  const initialValues = {
+    [FIELD_CREATE_SUGGESTED_SAFE_NAME]: getRandomName('safe'),
```

Comment:
If you call it as a function, it will give a new random name on each render. Please use the `useMnemonicSafeName` hook.

Reply:
good catch! done!